code/software/libs/src/start_serial/link_scripts/hugerom_rosco_m68k_program.ld

```diff
@@ -0,0 +1,135 @@
+/*
+ *------------------------------------------------------------
+ *                                  ___ ___ _
+ *  ___ ___ ___ ___ ___       _____|  _| . | |_
+ * |  _| . |_ -|  _| . |     |     | . | . | '_|
+ * |_| |___|___|___|___|_____|_|_|_|___|___|_,_|
+ *                     |_____|     firmware v1.2
+ * ------------------------------------------------------------
+ * Copyright (c)2020 Ross Bamford
+ * See top-level LICENSE.md for licence information.
+ *
+ * Link script for code that is loaded by the serial-receive
+ * boot loader.
+ * ------------------------------------------------------------
+ */
+
+OUTPUT_FORMAT("binary")
+ENTRY(START)
+
+ram  = 0x00002000;          /* start of user memory                     */
+
+MEMORY
+{
+  RAM  : org = ram,  l = 0x00f7e000     /* program can use all RAM > 8K */
+}
+
+/* program configuration symbols that may be useful to override */
```

Comment:
Any thoughts on splitting out the ROM-independent code from these, and including that with an `INCLUDE`? It might make it easier to keep everything in sync.

Reply:
The linker symbols have the advantage that they are really the only way to have absolute addresses from C (with simple extern).  Also handy from asm, but less of an issue there.  It might be nice to have the firmware revision in the linker ld name though (since it is a "matched set").

Reply:
This is probably a good idea @0xTJ , I guess we could move the common stuff out to a generic link script that then gets INCLUDED in the specific ones (which could then have the revision in the filename too as @XarkLabs suggests). 

Worth looking into I think. Would you mind raising an issue for it?